duckduckgo/iOS
# Improve Bookmarks DB Setup

Task/Issue URL: https://app.asana.com/0/856498667320406/1207875839908777/f
- https://app.asana.com/0/856498667320406/1206775525709653/f

**Description**:

Tweak Database setup code to better detect abnormal conditions. Remove old pixel.



**Steps to test this PR**:

Perform clean install tests - if stack is created as expected with no weird Pixels/errors.

Perform app update tests - if stack is opened as expected with no weird Pixels/errors.

Perform out of space tests:
1. Run the app in simulator - figure out Bookmarks <db location> by looking at logs (
2. Create 1MB disk image: `hdiutil create -size 1m -fs HFS+ 1MB.dmg`
3. Go to <db location> and copy contents to some outside directory.
4. Execute `hdiutil attach 1MB.dmg -mountpoint <db location>` to mount 1MB image over the container.
5. Copy data from point 3 back into the container.
6. Fill up the container with random files.

With this setup you should be able to see how app behaves when you restart it. 


**Definition of Done (Internal Only)**:


**Device Testing**:


**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
